src/components/Panels/Guider/Diagram/Diagram.tsx

```diff
@@ -239,7 +238,8 @@ function Flow() {
 
     setNodes([...sourceNodes, ...initialNodes]);
     setEdges([...sourceEdges, ...initialEdges]);
-    // setTimeout(() => fitView(), 100)
+
+    setTimeout(() => fitView({ duration: 1000 }), 20);
```

Comment:
Isn't this duration a bit high? 

Also for timeouts you should use the cleanup function from useEffect to clear the timeout:

```js
const timeout = setTimeout(...);

return () => clearTimeout(timeout);
```

Reply:
Totally agree about the cleanup.
The duration time is for the animation, at the beginning I was using 200 ms, I know short times are suggested to improve the usability and quick interactions, but I tested few times and 1 sec is nice to draw the attention of the user to see what's happening with the diagram.